x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsPendingDeletionsIntegTests.java

```diff
@@ -0,0 +1,575 @@
+                    snapshotInfo.snapshot().getSnapshotId().getName(),
+                    randomFrom(snapshotInfo.indices()),
+                    Settings.builder().put(SEARCHABLE_SNAPSHOTS_DELETE_SNAPSHOT_ON_INDEX_DELETION, true).build(),
+                    Strings.EMPTY_ARRAY,
+                    true,
+                    randomFrom(MountSearchableSnapshotRequest.Storage.values())
+                ),
+                new ActionListener<>() {
+                    @Override
+                    public void onResponse(RestoreSnapshotResponse response) {
+                        assertThat(response.getRestoreInfo().successfulShards(), greaterThan(0));
+                        assertThat(response.getRestoreInfo().failedShards(), equalTo(0));
+                        mounts.put(mount, snapshotInfo.snapshotId());
+                        mountLatch.countDown();
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        final AssertionError error = new AssertionError("error during mount", e);
+                        logger.error("test failed", error);
+                        mountLatch.countDown();
+                        throw error;
+                    }
+                }
+            );
+        }
+        mountLatch.await();
+
+        if (randomBoolean()) {
+            // we can't delete a repository that is used by searchable snapshot indices but we can update it
+            assertAcked(
+                clusterAdmin().preparePutRepository(repository)
+                    .setType(FsRepository.TYPE)
+                    .setVerify(randomBoolean())
+                    .setSettings(randomBoolean() ? repositorySettings : repositorySettings.put("dummy", randomInt()))
+            );
+        }
+
+        final CyclicBarrier startThreads = new CyclicBarrier(2);
+        final Thread deleteAllSnapshotsThread = new Thread(() -> {
+            try {
+                startThreads.await();
+                final CountDownLatch latch = new CountDownLatch(snapshots.size());
+                for (SnapshotInfo snapshot : snapshots) {
+                    client().admin()
+                        .cluster()
+                        .prepareDeleteSnapshot(repository, snapshot.snapshotId().getName())
+                        .execute(ActionListener.wrap(latch::countDown));
+
+                }
+                latch.await();
+            } catch (Exception e) {
+                throw new AssertionError(e);
+            }
+        });
+        deleteAllSnapshotsThread.start();
+
+        final Thread deleteAllMountedIndicesThread = new Thread(() -> {
+            try {
+                ensureGreen("mount-*");
+                startThreads.await();
+                final Set<String> mountedIndices = new HashSet<>(mounts.keySet());
+                do {
+                    List<String> deletions = randomSubsetOf(Math.max(1, mountedIndices.size() - 1), mountedIndices);
```

Comment:
Could we randomly pick the size between 1 and `mountedIndices.size()` instead? That would sometimes do them all in one go and sometimes in multiple rounds rather than always two rounds when `mountedIndices.size() > 1`?

Reply:
Yes, that makes more sense => ad62175111bfbf16472c604c66e44520ce90fb9a